apps/cowswap-frontend/src/modules/limitOrders/hooks/useHandleOrderPlacement.ts

```diff
@@ -124,8 +150,59 @@ export function useHandleOrderPlacement(
     analytics,
   ])
 
+  // Enhanced trade function with GTM tracking
+  const enhancedTradeFn = useCallback(async () => {
+    try {
+      // Track order submission
+      if (account) {
+        tradeTracking.onOrderSubmitted({
+          walletAddress: account,
+          tradeType: TradeType.LIMIT,
+          fromAmount: inputCurrencyInfo.amount ? parseFloat(inputCurrencyInfo.amount.toSignificant(6)) : undefined,
+          fromCurrency: inputCurrencyInfo.symbol,
+          fromAmountUSD: inputCurrencyInfo.fiatAmount ?? undefined,
+          toAmount: outputCurrencyInfo.amount ? parseFloat(outputCurrencyInfo.amount.toSignificant(6)) : undefined,
+          toCurrency: outputCurrencyInfo.symbol,
+          toAmountUSD: outputCurrencyInfo.fiatAmount ?? undefined,
+          contractAddress,
+        })
+      }
+
+      // Execute the trade
+      const result = await tradeFn()
+
+      return result
+    } catch (error) {
+      // Track failure
+      if (account) {
+        tradeTracking.onOrderFailed(
+          {
+            walletAddress: account,
+            tradeType: TradeType.LIMIT,
+            fromCurrency: inputCurrencyInfo.symbol,
+            toCurrency: outputCurrencyInfo.symbol,
+            contractAddress,
+          },
+          error instanceof Error ? error.message : String(error),
+        )
+      }
+      throw error
+    }
+  }, [
+    account,
+    contractAddress,
+    inputCurrencyInfo.amount,
```

Comment:
You use every field of inputCurrencyInfo in the hook, why do you need to put them separately to the dependencies here?